backend/initialRecommendations.js

```diff
@@ -0,0 +1,230 @@
+  const priceScore =
+    1 - Math.abs(business.priceLevel - userPreferences.avgPriceLevel) / 4;
+  const distanceScore = 1 - userPreferences.avgDistance / 50;
+  const ratingScore = business.averageRating / 5;
+
+  const score =
+    subsequentWeights.priceLevel * priceScore +
+    subsequentWeights.location * distanceScore +
+    subsequentWeights.rating * ratingScore;
+
+  return score;
+}
+
+// Provide initial recommendations based on user's favorite categories, location, and preferred rating
+async function provideInitialRecommendations(user) {
+  const businesses = await fetchBusinessData();
+
+  const userPreferences = {
+    favoriteCategories: user.favoriteCategories,
+    coordinates: parseLocation(user.location),
+    preferredRating: user.preferredRating,
+  };
+
+  if (!userPreferences.coordinates) {
+    console.error("Invalid user location:", user.location);
+    return [];
+  }
+
+  const scoredBusinesses = businesses.map((business) => {
+    const score = scoreBusinessInitial(business, userPreferences);
+    return { ...business, score };
+  });
+
+  scoredBusinesses.sort((a, b) => b.score - a.score);
+
+  return scoredBusinesses.slice(0, 20);
+}
+
+// Recommend businesses based on user preferences
+async function recommendBusinesses(userId) {
+  const user = await prisma.user.findUnique({
+    where: { id: userId },
+    include: { interactions: true },
+  });
+  if (!user) {
+    console.error("User not found");
+    return [];
+  }
+
+  const interactions = await fetchLikedInteractions(userId);
+
+  // If there are no interactions or the user has liked fewer than 3 businesses, provide initial recommendations
+  if (!interactions || interactions.length < 3) {
+    return await provideInitialRecommendations(user);
+  }
+
+  // Analyze user preferences based on liked interactions
+  const userPreferences = analyzeUserPreferences(interactions);
+  if (!userPreferences) {
+    return await provideInitialRecommendations(user);
+  }
+
+  const businesses = await fetchBusinessData();
+
+  const scoredBusinesses = businesses.map((business) => {
+    const score = scoreBusinessSubsequent(business, userPreferences);
+    return { ...business, score };
+  });
+
+  scoredBusinesses.sort((a, b) => b.score - a.score);
+
+  return scoredBusinesses.slice(0, 20);
+}
+
+// Utility functions
+
+// Function to parse location into latitude and longitude
+const parseLocation = (location) => {
+  if (!location) return null;
+  const [lat, lng] = location
+    .split(",")
+    .map((coord) => parseFloat(coord.trim()));
+  if (isNaN(lat) || isNaN(lng)) return null;
+  return { lat, lng };
+};
+
+// Calculate Haversine distance between two points (latitude and longitude)
+const haversineDistance = (lat1, lon1, lat2, lon2) => {
+  const toRadians = (degree) => degree * (Math.PI / 180);
+  const R = 3958.8; // Radius of the Earth in miles
+  const dLat = toRadians(lat2 - lat1);
+  const dLon = toRadians(lon1 - lon2);
+  const a =
```

Comment:
Quick note that haversine distance will calculate shortest distance between two points which is a straight line; but travel distance can be actually longer because of freeways, intersections etc. I don't expect you to have accurate distances (seems out of scope) but it's something to note.